Library/Homebrew/untap.rb

```diff
@@ -0,0 +1,56 @@
+# typed: true
+# frozen_string_literal: true
+
+module Homebrew
+  # Helpers for the `brew untap` command.
+  # @api private
+  module Untap
+    # All installed formulae currently available in a tap by formula full name.
+    sig { params(tap: Tap).returns(T::Array[Formula]) }
+    def self.installed_formulae_for(tap:)
+      tap.formula_names.filter_map do |formula_name|
+        next unless installed_formulae_names.include?(T.must(formula_name.split("/").last))
+
+        formula = begin
+          Formulary.factory(formula_name)
+        rescue
+          # Don't blow up because of a single unavailable formula.
+          next
+        end
```

Comment:
I'm more comfortable swallowing these errors now that we have some tests to sanity check that things are working as expected.